### ISSUE TITLE
Options for coarser-grained "auto" group-by time

### ISSUE THREAD

**Issue by ryan-williams** [grafana/grafana#3291]
I have some series graphed as bars, with "group by" set to `auto` and "group by time interval" set to `>1s` because my data was recorded every second.

When zoomed out, I get the faintest, barely visible bars:

![](http://cl.ly/image/0J0B2n2g3623/Screen%20Shot%202015-11-19%20at%209.10.13%20PM.png)

Really what I want it to look like is what I get when I set the "group by time interval" to `>1m`:

![](http://cl.ly/image/0S3i3r2j0d43/Screen%20Shot%202015-11-19%20at%209.13.37%20PM.png)

But I want to be able to zoom way in and automatically get a sane number of bars again, probably 50-100 total from my examples here.

If I zoom in with `>1m` set, of course I end up with a silly picture, with bars too wide:

![](http://cl.ly/image/1b2O3T050F2K/Screen%20Shot%202015-11-19%20at%209.15.02%20PM.png)

At this zoom level I want the group-by to be more granular, e.g. `>1s`:

![](http://cl.ly/image/2p3q1d0a2F3e/Screen%20Shot%202015-11-19%20at%209.15.28%20PM.png)

So basically: I want the ability to tell the "auto group-by" how many intervals to slice my panel's time-range into. It seems to be doing >600 in the first image above (every 3-4s over a 30m window), but I want it to do a fraction of that, more like 100-300, from my second and fourth images here.

**Comment by torkelo** [comment]
create a template variable of the type interval, for this variable you can specify how many steps its 'auto' options should be. Use the variable in the group by time field.
